R/SingleR.visualizations.R

```diff
@@ -0,0 +1,121 @@
+#' Plot a scatter plot of a single cell vs. a single reference cell/sample
+#'
+#' @param sc_data Similar to SinlgeR test input, a numeric matrix of single-cell expression values (usually
+#' log-transformed or otherwise variance-stabilized), where rows are genes and columns are cells.
+#' Alternatively, a \linkS4class{SingleCellExperiment} object containing such a matrix.
+#' @param cell_id a number indicating which single cell to use.
+#' Alternatively, the name of the cell.
+#' @param refdata Similar to SinlgeR training input, a numeric matrix of expression values. 
+#' Alternatively, a \linkS4class{SingleCellExperiment} object containing such a matrix.
+#' This should have the same rows as or a subset of the rows in \code{test}.
+#' @param sample_id a number of the sample to use
+#' @param assay.type An integer scalar or string specifying the assay of \code{sc_data} or \code{refdata} containing the relevant expression matrix,
+#' if \code{sc_data} or \code{refdata} is a \linkS4class{SingleCellExperiment} object.
+#'
+#' @return a gpplot scatterplot of the expression of indidivual genes, with linear regression and Spearman pariwaise correlation coefficient
+#' 
+#' @export
+#' @importFrom ggplot2 ggplot geom_point geom_smooth theme xlab ylab ggtitle theme_classic
+SingleR.DrawScatter = function(sc_data, cell_id, refdata,sample_id, assay.type = 1) {
+  if (is(sc_data, "SingleCellExperiment")) {
+    sc_data <- assay(sc_data, assay.type)
+  }
+  if (is(refdata, "SingleCellExperiment")) {
+    refdata <- assay(refdata, assay.type)
+  }
+  if (is(refdata, "list")) {
+    refdata <- refdata$data
+  }
+  rownames(sc_data) = tolower(rownames(sc_data))
```

Comment:
Use `<-` instead of `=`. The former is more amenable to static code analysis, e.g., when you want to know assignments to a variable name `x`, you can just look for `x <-` without worrying about calls to functions like `A(x=1)`.

Reply:
Done.  Was a remnant from the old code.